### PR TITLE
DTSPO-9914 Automate updating of tfcmt, pester, tfenv

### PR DESCRIPTION
**Pull request by Tyler-34**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-9914

### Change description ###
Adding workflows to automatically create pull requests to update the versions for pester, tfcmt and tfenv.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .github/workflows/tfenv-updatecli.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          version: v0.33.3
+
+      - uses: tibdex/github-app-token@v1
+        id: generate-token
+        with:
+          app_id: ${{ secrets.DEPENDENCY_UPDATER_APP_ID }}
+          private_key: ${{ secrets.DEPENDENCY_UPDATER_APP_PRIVATE_KEY }}
+
+      - name: Run Updatecli in Dry Run mode
+        run: updatecli diff --config ./updatecli/tfenv-weekly.d --values ./updatecli/values.github-action.yaml
```
Comment (timja) [inline .github/workflows/tfenv-updatecli.yaml:33]: do we need a different action for each? I thought updatecli would take a directory of files and process them all creating a PR for each

e.g. see how it's used here:
https://github.com/jenkins-infra/jenkins-infra/blob/b8493cb24d9e6f68be8cc8dcee5f9fc260526a83/.github/workflows/updatecli.yaml#L27

https://github.com/jenkins-infra/jenkins-infra/tree/production/updatecli/weekly.d

**Review by timja — CHANGES_REQUESTED** [review]
I don't think we need 3 GitHub actions, the rest looks good to me.

File: .github/workflows/tfenv-updatecli.yaml (hunk 1 above)
Comment (Tyler-34) [reply .github/workflows/tfenv-updatecli.yaml]: You are correct doesn't look like we need workflow action for each config.  I have made changes so only one workflow is present.

File: updatecli/weekly.d/tfcmt.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  setTfcmtImageVersion:
+    kind: github
+    scmid: default
+    title: >-
+      [updatecli] Bump Tfcmt version to {{ source "latestTfcmtImageVersion" }}
+    spec:
+      automerge: false
+      draft: false
+      description: |
+        Bump Tfcmt version to {{ source "latestTfcmtImageVersion" }}
```
Comment (timja) [inline updatecli/weekly.d/tfcmt.yaml:50]:
```suggestion
        Bump Tfcmt version to {{ source "latestTfcmtImageVersion" }}

```

File: updatecli/weekly.d/pester.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  setPesterImageVersion:
+    kind: github
+    scmid: default
+    title: >-
+      [updatecli] Bump pester version to {{ source "latestPesterImageVersion" }}
+    spec:
+      automerge: false
+      draft: false
+      description: |
+        Bump pester version to {{ source "latestPesterImageVersion" }}
```
Comment (timja) [inline updatecli/weekly.d/pester.yaml:50]:
```suggestion
        Bump pester version to {{ source "latestPesterImageVersion" }}

```

File: updatecli/weekly.d/tfenv.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        - scripts/tfenv-install-terraform.sh
+      matchpattern: 'git clone -b v(\d+\.)?(\d+\.)?(\*|\d+)'
+      content: 'git clone -b v{{ source `latestTfenvImageVersion` }}'
+
+pullrequests:
+  setTfenvImageVersion:
+    kind: github
+    scmid: default
+    title: >-
+      [updatecli] Bump Tfenv version to {{ source "latestTfenvImageVersion" }}
```
Comment (timja) [inline updatecli/weekly.d/tfenv.yaml:45]: it's tfenv not Tfenv (applies throughout)

```suggestion
      [updatecli] Bump tfenv version to {{ source "latestTfenvImageVersion" }}
```

File: updatecli/weekly.d/tfenv.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    spec:
+      automerge: false
+      draft: false
+      description: |
+        Bump Tfenv version to {{ source "latestTfenvImageVersion" }}
```
Comment (timja) [inline updatecli/weekly.d/tfenv.yaml:50]:
```suggestion
        Bump Tfenv version to {{ source "latestTfenvImageVersion" }}

```

**Review by timja — COMMENTED** [review]
looks good apart from some minor touchups

**Pull request closed without merging**